src/py4vasp/calculation/_partial_charge.py

```diff
@@ -137,6 +147,7 @@ def to_stm(
             object.
         """
         _raise_error_if_vacuum_too_small(self._estimate_vacuum())
+        self.stm_settings = stm_settings
```

Comment:
Do I understand correctly that I should pass the `stm_settings` input along to the other methods as additional input instead of storing it in `self.stm_settings`?

I thought it would be a good idea to have a `stm_settings` property, so the user can query the current settings (including the defaults) by
`calc.partial_charge.stm_settings`

How would I do that if I do not store the settings?

Reply:
If you want to access the settings that a STM image has been generated with, you should store it in the returned image. I think you should be able to just set an attribute on the Contour class.
~~~python
graph = calc.partial_charge.to_stm()
graph.series[0].settings
~~~
Another alternative that would be perhaps be more modular is to add a meta_data attribute to the Graph class. Then you could use
~~~python
graph = calc.partial_charge.to_stm()
graph.meta_data["stm_settings"]
~~~

Reply:
You can still get the default settings with the `settings` property, it is just not a good idea to allow mutating that because it leads to hard to spot errors.

Reply:
A complete use case could then look like
~~~python
from py4vasp import calculation as calc
stm_settings = calc.partial_charge.stm_settings
# ... modify stm settings
graph = calc.partial_charge.to_stm(..., stm_settings)
~~~